### PR TITLE
Doc: Add quickstart

### PR DESCRIPTION
**Pull request by matthiasdiener**
I'd be interested in knowing if this is worthwhile to add to the manual.

Here is how it looks like rendered: https://charm.readthedocs.io/en/add-quickstart/quickstart.html

### CODE REVIEW

File: doc/quickstart.rst [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+
+To build Charm++, use the following commands:
+
+.. code-block:: console
+
+   $ cd charm
+   $ ./build AMPI netlrts-linux-x86_64 --with-production -j4
+
+This is the recommenced version to install Charm++ on Linux systems.
```
Comment (stwhite91) [inline doc/quickstart.rst]: maybe say "to install Charm++ and AMPI"

Comment (matthiasdiener) [reply doc/quickstart.rst]: Thats a good point. Should we keep this to build AMPI, or change to charm++/LIBS?

File: doc/quickstart.rst [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+For MacOS, substitute "linux" with "darwin". For advanced compilation options,
+please see Section :numref:`sec:install` of the manual.
+
+
+Parallel "Hello World" with Charm++
+-----------------------------------
+
+The basic unit of computation in Charm++ is a **chare**, which is a C++
+object. Chares have **entry methods** that can be invoked asynchronously.
+in A Charm++ application consists of collections of chares (such as chare arrays)
```
Comment (nitbhat) [inline doc/quickstart.rst]: In a Charm++

Comment (nitbhat) [reply doc/quickstart.rst]: Actually, looking at the sentence, there should be no "in". It should just be "A Charm++ application consists"

Comment (matthiasdiener) [reply doc/quickstart.rst]: fixed

File: doc/quickstart.rst (hunk 1 above)
Comment (matthiasdiener) [reply doc/quickstart.rst]: changed to charm++.

File: doc/quickstart.rst [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+   class Hello : public CBase_Hello 
+   {
+   public:
+     Hello() {}
+
+     void SayHi()
+     {
+       CkPrintf("PE %d says: Hello world from element %d.\n", CkMyPe(), thisIndex);
+       if (thisIndex < nElements-1) {
+         thisProxy[thisIndex+1].SayHi(); // Pass the hello on
```
Comment (rbuch) [inline doc/quickstart.rst]: A minor stylistic point, but I think the - and + operators should have spaces around them.

File: doc/quickstart.rst [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+   PE 0 says Hi from element 3
+   PE 1 says Hi from element 4
+   PE 1 says Hi from element 5
+   PE 1 says Hi from element 6
+   PE 1 says Hi from element 7
+   All done
+   [Partition 0][Node 0] End of program
+
+
+Where to go from here
```
Comment (rbuch) [inline doc/quickstart.rst]: This should be title cased for consistency.

**Comment by rbuch** [comment]
Since this is intended to be one of the first things people do, should it have explanations of what various things in the source code are (like that `entry <something>` registers an remotely invocable method or that `ckNew` creates a chare array of size `nElements`)? I understand that this is supposed to be concise and that users can look to the rest of the documentation if they want more information, but I think a spot of commenting would be helpful.

File: doc/quickstart.rst (hunk 4 above)
Comment (matthiasdiener) [reply doc/quickstart.rst]: Done

File: doc/quickstart.rst (hunk 3 above)
Comment (matthiasdiener) [reply doc/quickstart.rst]: Done

**Comment by matthiasdiener** [comment]
> Since this is intended to be one of the first things people do, should it have explanations of what various things in the source code are (like that `entry <something>` registers an remotely invocable method or that `ckNew` creates a chare array of size `nElements`)? I understand that this is supposed to be concise and that users can look to the rest of the documentation if they want more information, but I think a spot of commenting would be helpful.

I added a few more comments in the code that should make entry methods and ckNew more clear.

**Comment by matthiasdiener** [comment]
This is ready for review.

File: doc/quickstart.rst [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+- The ``tests/charm++/simplearrayhello`` folder in the Charm++ distribution has a more comprehensive example, from
+  which the example in this file was derived.
+
+- The main Charm++ manual (https://charm.readthedocs.io/) contains more information about developing
+  and running Charm++ applications.
+
+- Charm++ has lots of other features, such as chare migration, load balancing,
+  and checkpoint/restart. The main manual has more information about them.
+
+- AMPI is an implementation of MPI on top of Charm++.
```
Comment (nitbhat) [inline doc/quickstart.rst]: Maybe add a link to the AMPI manual here? (https://charm.readthedocs.io/en/latest/ampi/manual.html)

Comment (matthiasdiener) [reply doc/quickstart.rst]: done

**Pull request merged**